pandas/tests/indexing/test_interval.py

```diff
@@ -192,11 +295,63 @@ def test_non_unique_moar(self):
         with pytest.raises(ValueError):
             s[[Interval(1, 3)]]
 
+    @pytest.mark.xfail(reason="new indexing tests for issue 16316")
+    def test_non_unique_moar_updated_behavior(self):
+
+        idx = IntervalIndex.from_tuples([(1, 3), (1, 3), (3, 7)])
+        s = Series(range(len(idx)), index=idx)
+
+        expected = s.iloc[[0, 1]]
+        result = s.loc[Interval(1, 3)]
+        tm.assert_series_equal(expected, result)
+
+        # non-unique index and slices not allowed
+        with pytest.raises(ValueError):
+            s.loc[Interval(1, 3):]
+        # this is confusing to me. I would have done:
+        # expected = s
+        # result = s.loc[Interval(1, 3):]
+        # tm.assert_series_equal(expected, result)
```

Comment:
This does not seem to be fully true. As long as they are still sorted, slicing with duplicates seems to work (at least for this test case):

```
In [36]: s = pd.Series(range(3), index=[1,1,2])

In [37]: s.loc[1:]
Out[37]: 
1    0
1    1
2    2
dtype: int64
```